src/apps/competitions/models.py

```diff
@@ -23,6 +28,107 @@ def __str__(self):
     def bundle_dataset(self):
         return CompetitionCreationTaskStatus.objects.get(resulting_competition=self).dataset
 
+    def check_future_phase_submissions(self):
+        """
+        Checks for if we need to migrate current phase submissions to next phase.
+        """
+        phases = self.phases.all()
+
+        if len(phases) == 0:
+            return
+
+        try:
+            current_phase = phases.get(status='Current')
+            next_phase = phases.get(status='Next')
+            final_phase = phases.get(status='Final')
+
+        except Phase.DoesNotExist:
+            return
+
+        # Making sure current_phase or next_phase is not None
+        if current_phase is None or next_phase is None or current_phase is final_phase:
+            return
+
+        logger.info("Checking for needed migrations on competition pk=%s, current phase: %s, next phase: %s" %
+                    (self.pk, current_phase.index, next_phase.index))
+
+        # Check for next phase and see if it has auto_migration enabled
+        if next_phase and next_phase.auto_migration:
+            if next_phase.start < now() and current_phase.end < now():
+                self.apply_phase_migration(current_phase, next_phase)
+
+    def apply_phase_migration(self, current_phase, next_phase):
+        '''
+        Does the actual migrating of submissions from current_phase to next_phase
+
+        :param current_phase: The phase object to transfer submissions from
+        :param next_phase: The new phase object we are entering
+        '''
+        logger.info("Checking for submissions that may still be running competition pk=%s" % self.pk)
+
+        if current_phase.submissions.filter(status='Scoring').exists():
```

Comment:
The status check here should include more statuses, here are all the options currently:

```
    CANCELLED = "Cancelled"
    FINISHED = "Finished"
    FAILED = "Failed"
```

I think may be best to instead have a `completed_when` field on submissions? (what do you think about that naming, along with `created_when`...) I made a new issue for this here https://github.com/codalab/competitions-v2/issues/203

For now check if there are any submissions NOT "finished, failed, or cancelled"

Reply:
Resolved in latest push